### ISSUE TITLE
wait-for-status triggered on pull_request fails to detect failed jobs

### ISSUE THREAD

**Issue by Crow-EH** [WyriHaximus/github-action-wait-for-status#21]
**[UPDATE]:** Since 5786ef2, wait-for-status triggered on pull_request does not detect failed jobs. See  https://github.com/WyriHaximus/github-action-wait-for-status/issues/21#issuecomment-670522475

Hi !

Since we upgraded from `d2ddfe5` to `v1` all our check jobs finish in success with a weird message (since other jobs are already in progress) : `No statuses found, assuming success`.

I would have linked one of our jobs here, but since it also happens in the CI here, juste see your CI logs :smile: :
https://github.com/WyriHaximus/github-action-wait-for-status/runs/907377690?check_suite_focus=true

It seems pretty weird to me. Is it intended ?

PS: Still love your action anyway :heart:

**Comment by WyriHaximus** [comment]
Look at https://github.com/ekino/docker-buildbox/pull/313 you are only using Github Actions correct? So within the Github API there are two kinds of commit "statuses" the older `statuses` and the newer `checks`. Generally, that message means it couldn't find any `statuses` and just assumes those are fine, but it doesn't ignore any `checks`. I'll see how I can improve this 👍 , thank you for reporting

**Comment by Crow-EH** [comment]
Ok I get the distinction.

We still have a problem anyway : It also pass even when GitHub Actions checks fail.

See this check suite for example : https://github.com/ekino/docker-buildbox/pull/316/checks?check_run_id=948414227

(taking a screenshot for posterity)

Here we have `check` in success, when it should have failed because of the `php` jobs 
![image](https://user-images.githubusercontent.com/3366767/89514214-db91e900-d7d5-11ea-816b-6eff4bc3ccdd.png)

It used to work (fail successfully) before upgrading.

Here is our check action for reference :
```
name: Build Checker

on:
  pull_request:
    branches:
      - master

jobs:
  check:
    runs-on: ubuntu-latest
    steps:
      - name: Wait for commit statuses
        id: status
        uses: WyriHaximus/github-action-wait-for-status@v1
        with:
          ignoreActions: check
          checkInterval: 60
        env:
          GITHUB_TOKEN: "${{ secrets.GITHUB_TOKEN }}"
      - name: Succeed
        if: steps.status.outputs.status == 'success'
        run: exit 0
      - name: Fail
        if: steps.status.outputs.status == 'failure'
        run: exit 1
```

**Comment by WyriHaximus** [comment]
Ok cool thanks, will have a look at that. Also if you have a simple reproduction project, even a commit would be enough to test against that would be amazing :+1:

**Comment by Crow-EH** [comment]
Alright I got one : It seems to be working correctly on push events (failing job on push + check on push), but not on pull request.

Atm, I only have a failing job on PR + check on PR here : https://github.com/Crow-EH/github-action-wait-for-status-issue-21/pull/1/checks?check_run_id=958462978

I'll try a failing job on push + check on PR asap.

**Comment by Crow-EH** [comment]
Same issue with a check action on PR + a failing job on push : https://github.com/Crow-EH/github-action-wait-for-status-issue-21/pull/2/checks?check_run_id=958480494

The wait-for-status seems to fail to see the failing job when ran on a PR event.

**Comment by Crow-EH** [comment]
WIP

OK: The check action correctly detects the failing job, then fails as intended.
NOK: The check action assumes success.

Version `d2ddfe5`:
* PR/PR OK :heavy_check_mark:  : https://github.com/Crow-EH/github-action-wait-for-status-issue-21/pull/3/checks?check_run_id=958550489
* PR/Push OK :heavy_check_mark:  : https://github.com/Crow-EH/github-action-wait-for-status-issue-21/pull/4/checks?check_run_id=958555869

**Comment by Crow-EH** [comment]
There's a lot of changes in 5786ef2, so I don't really know where to search, but I guess you do @WyriHaximus. :smile: 

I hope I've helped a little.

**Comment by Crow-EH** [comment]
In case you're wondering, the bug is also happening when the failing job and wait-for-status are in the same action : https://github.com/Crow-EH/github-action-wait-for-status-issue-21/pull/11/checks?check_run_id=958722990

**Comment by WyriHaximus** [comment]
@Crow-EH absolutely awesome write up, I'll have a long look at it tomorrow :+1:

**Comment by WyriHaximus** [comment]
@Crow-EH So a few interesting things came up. First off, now all commits have checks, but do have a failure on the PR so that could be a new feature to look into. Instead of check the commit status. Check commit status on push events and pull_request status on PR events. Secondly I'm adding additional login (via #22) and release that as v1.1 later today.

**Comment by WyriHaximus** [comment]
@Crow-EH OK v1.1 is out. Can you run against that for me? Interested in the output. Looking into the PR vs push difference next

**Comment by Crow-EH** [comment]
Will do ASAP

**Comment by Crow-EH** [comment]
Still not working in 1.1 : https://github.com/Crow-EH/github-action-wait-for-status-issue-21/pull/14/checks

Though I think I found the main difference between Push and PR events : The checked commit(s).

## Last working version (4c9e588) on PR
https://github.com/Crow-EH/github-action-wait-for-status-issue-21/pull/8/checks?check_run_id=958608707

Actual commits:
![image](https://user-images.githubusercontent.com/3366767/89766443-72bdb000-daf8-11ea-81f9-bc69565d55ea.png)

Commit that should be checked:
![image](https://user-images.githubusercontent.com/3366767/89766509-8c5ef780-daf8-11ea-8d9f-ced59868cf22.png)

Logs:
```
Checking statuses and checks for commit: 2953b4d8b12c7dec6a06dd4d73d29744d085902b
Checking statuses and checks for commit: bf576007f9159dc766e70ed6eb747da637175ef0
Checking statuses and checks for commit: c077f12b6dcce69862995e5edd60023f0ce73c10
```

✔️Working

Hypothesis: While it uselessly check the parent commit (maybe it is? I don't know.) and the first commit, it also checks the last commit so it works as intended.


## First failing version (775e5b0) on Push
https://github.com/Crow-EH/github-action-wait-for-status-issue-21/pull/16/checks?check_run_id=965963048

Actual commits:
![image](https://user-images.githubusercontent.com/3366767/89767094-861d4b00-daf9-11ea-9d04-6c50dab17625.png)

Commit that should be checked:
![image](https://user-images.githubusercontent.com/3366767/89767124-9503fd80-daf9-11ea-85eb-f288e4982656.png)

Logs:
```
Checking statuses and checks for commit: b6c8658edcf491fedeea39efa12fb62316aa2345
```

✔️Working

Hypothesis: It only checks the last commit so it works as intended. It may be the only commit listed since it's a Push event ?


## First failing version (775e5b0) on PR
https://github.com/Crow-EH/github-action-wait-for-status-issue-21/pull/10/checks?check_run_id=958630684

Actual commits:
![image](https://user-images.githubusercontent.com/3366767/89766681-d3e58380-daf8-11ea-91fd-9d796563527f.png)

Commit that should be checked:
![image](https://user-images.githubusercontent.com/3366767/89766702-dfd14580-daf8-11ea-85cc-3d3f87a233f5.png)

Logs:
```
Checking statuses and checks for commit: ccac34cd098f714b6e6b363f8fdcc03d81a66622
```

❌Not Working

Hypothesis: It only checks the first commit so it doesn't work.

**Comment by WyriHaximus** [comment]
> Still not working in 1.1 : https://github.com/Crow-EH/github-action-wait-for-status-issue-21/pull/13/checks?check_run_id=965850322
> 
> ```
> [2020-08-10 08:31:13.939919] wait.DEBUG: Iterating over 0 check(s)
> [2020-08-10 08:31:13.939976] wait.DEBUG: All checks completed, marking resolve and success
> ```

Ok cool as expected! I'll start on PR detection then. Thanks :+1:

**Comment by WyriHaximus** [comment]
@Crow-EH ok I think I've figured it out. 

* So this PR is the basis: https://github.com/Crow-EH/github-action-wait-for-status-issue-21/pull/13
* This commit is the only commit in that PR: https://github.com/Crow-EH/github-action-wait-for-status-issue-21/pull/13/commits/9019aec1c0ef4c367eac85cd9aaebe6fa45922b0
* Now this action is ran in the following check https://github.com/Crow-EH/github-action-wait-for-status-issue-21/runs/965850322 which checks the status of this commit https://github.com/Crow-EH/github-action-wait-for-status-issue-21/commit/99f5b351cc76b397ff0f48f312f5b16b40f74449 instead of the latest commit of the PR as we expect it to
* When looking at the API it does refer to the commit we expect as a parent commit https://api.github.com/repos/Crow-EH/github-action-wait-for-status-issue-21/commits/99f5b351cc76b397ff0f48f312f5b16b40f74449
* Which :drum: :drum: :drum: is the `merge_commit_sha` of that PR when looking at the API https://api.github.com/repos/Crow-EH/github-action-wait-for-status-issue-21/pulls/13

TL;DR when detecting it's ran on a PR, I can get the `merge_commit_sha` from the PR and check against that when it's detected that the action is ran in a PR :tada:

**Comment by Crow-EH** [comment]
@WyriHaximus wow, that's great news!
And this new beahviour would be straight and simple too, which is nice.

Thanks a lot!

**Comment by WyriHaximus** [comment]
@Crow-EH it is, might have to check both commits to be sure but will see :D

**Comment by WyriHaximus** [comment]
@Crow-EH detection seems to be working: https://github.com/WyriHaximus/github-action-wait-for-status/runs/1038925268?check_suite_focus=true#step:4:133 WIP PR at #29

**Comment by Crow-EH** [comment]
Nice! Can't wait to be able to test it

**Comment by WyriHaximus** [comment]
Should be able to merge it this weekend

**Comment by WyriHaximus** [comment]
@Crow-EH go for it, [`v1.2(.0)`](https://github.com/WyriHaximus/github-action-wait-for-status/releases/tag/v1.2.0) has been tagged and it available immediately.

**Comment by Crow-EH** [comment]
@WyriHaximus It detects failing jobs correctly now 🎉

I'm closing this issue now, thank you. 😄 

Great job!

**Comment by WyriHaximus** [comment]
@Crow-EH Awesome to hear that, and you're welcome 🎉!
